spec/models/offline_transaction_refund_spec.rb

```diff
@@ -5,5 +5,232 @@
 require 'rails_helper'
 
 RSpec.describe OfflineTransactionRefund, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+	let!(:nonprofit) { create(:nm_justice) }
+	let!(:supporter) { force_create(:supporter, nonprofit: nonprofit) }
+
+	let(:offline_transaction_refund) do
+		build(
+			:offline_transaction_refund,
+			payment:
+				force_create(
+					:payment,
+					gross_amount: 500,
+					net_amount: 400,
+					fee_total: 100,
+					date: Time.zone.now,
+					nonprofit: nonprofit,
+					supporter: supporter
+				)
+		)
+	end
+	let(:offline_transaction_charge) do
+		build(
+			:offline_transaction_charge,
+			payment:
+				force_create(
+					:payment,
+					gross_amount: 400,
+					net_amount: 300,
+					fee_total: 100,
+					date: Time.zone.now,
+					nonprofit: nonprofit,
+					supporter: supporter
+				)
+		)
+	end
+
+	let(:offline_transaction) { build(:offline_transaction) }
+	let(:transaction) do
+		trx = supporter.transactions.build(amount: 500)
+		trx.build_subtransaction(
+			subtransactable: OfflineTransaction.new(amount: 500),
+			subtransaction_payments: [
+				build(:subtransaction_payment, paymentable: offline_transaction_refund),
+				build(:subtransaction_payment, paymentable: offline_transaction_charge)
+			]
+		)
+		trx.save!
+		trx
+	end
+
+	let(:attributes) { double }
+	let(:event_publisher) { double }
+
+	before do
+		allow(Houdini)
+			.to receive(:event_publisher)
+			.and_return(event_publisher)
+		allow(event_publisher)
+			.to receive(:announce)
+			.with(any_args)
+		transaction
+	end
+
+	after do
+		Timecop.return
+	end
```

Comment:
Timecop should only be used as part of a block: i.e.

```ruby
Timecop.freeze(2020, 5,4) do 
 # The current time in this block is Time.new(2020, 5, 4)
end
```

Using Timecop outside a block risks the Time not being reset someplace and the frozen time leaking into a test we don't want it in.

If you want to have all of your examples into a `describe` or `context` block to use a specific frozen time, you put use an `around` block inside the describe or context:

```ruby
describe Foo do
 around(:each) do |example|
    Timecop.freeze(2020, 4, 5) do
      example.run
    end
  end

  it { # time is Time.new(2020, 4, 5)}

  it { # time is Time.new(2020, 4, 5)}

  it { # time is Time.new(2020, 4, 5)}
end
```

Reply:
That's good to know :)